src/OpenLoco/Windows/ToolbarBottomEditor.cpp

```diff
@@ -1,15 +1,145 @@
+        widget_t& previous = self->widgets[widx::previous_frame];
+        widget_t& next = self->widgets[widx::next_frame];
+
+        if (EditorController::canGoBack())
+        {
+            Gfx::drawRect(ctx, previous.left + self->x, previous.top + self->y, previous.width(), previous.height(), 0x2000000 | 52);
+        }
+        Gfx::drawRect(ctx, next.left + self->x, next.top + self->y, next.width(), next.height(), 0x2000000 | 52);
+
+        self->draw(ctx);
+
+        if (EditorController::canGoBack())
+        {
+            Gfx::drawRectInset(ctx, previous.left + self->x + 1, previous.top + self->y + 1, previous.width() - 2, previous.height() - 2, self->colours[1], 0x30);
+        }
+        Gfx::drawRectInset(ctx, next.left + self->x + 1, next.top + self->y + 1, next.width() - 2, next.height() - 2, self->colours[1], 0x30);
+
+        Gfx::drawStringCentred(*ctx, (previous.right + next.left) / 2 + self->x, self->y + self->height - 12, Colour::opaque(self->colours[2]) | Colour::outline_flag, _stepNames[EditorController::getCurrentStep()]);
+
+        if (EditorController::canGoBack())
+        {
+            Gfx::drawImage(ctx, self->x + previous.left + 6, self->y + previous.top + 6, ImageIds::step_back);
+            int x = (previous.left + 30 + previous.right) / 2;
+            int y = previous.top + 6;
+            colour_t al = Colour::opaque(self->colours[1]);
+            if (Input::isHovering(self->type, self->number, widx::previous_button))
+            {
+                al = Colour::white;
+            }
+            Gfx::drawStringCentred(*ctx, self->x + x, self->y + y, al, StringIds::editor_previous_step);
+            Gfx::drawStringCentred(*ctx, self->x + x, self->y + y + 10, al, _stepNames[EditorController::getPreviousStep()]);
+        }
+        Gfx::drawImage(ctx, self->x + next.right - 29, self->y + next.top + 4, ImageIds::step_forward);
+        int x = next.left + (next.width() - 31) / 2;
+        int y = next.top + 6;
+        colour_t al = Colour::opaque(self->colours[1]);
+        if (Input::isHovering(self->type, self->number, widx::next_button))
+        {
+            al = Colour::white;
+        }
+        Gfx::drawStringCentred(*ctx, self->x + x, self->y + y, al, StringIds::editor_next_step);
+        Gfx::drawStringCentred(*ctx, self->x + x, self->y + y + 10, al, _stepNames[EditorController::getNextStep()]);
+    }
+
+    // 0x0043D0ED
+    static void onMouseUp(window*, widget_index i)
+    {
+        switch (i)
+        {
+            case widx::previous_button:
+                EditorController::goToPreviousStep();
+                break;
+
+            case widx::next_button:
+                EditorController::goToNextStep();
+                break;
+        }
+    }
+
     // 0x0043CCCD
     void open()
     {
-        call(0x0043CCCD);
-        // Note call to window_resize_gui_scenario_editor not part of this function
-        // move when implemented.
+        events.on_mouse_up = onMouseUp;
```

Comment:
I'd still make a separate function for setting the events to just make it consistent with the rest of the codebase.